bundles/mapping/mapmodule/plugin/layers/LayerSelectionPlugin.js

```diff
@@ -745,18 +776,29 @@ Oskari.clazz.define('Oskari.mapframework.bundle.mapmodule.plugin.LayerSelectionP
             var me = this;
 
             header.on('click', function () {
-                if (me.popup && me.popup.isVisible()) {
-                    me.popup.getJqueryContent().detach();
-                    me.popup.close(true);
-                    me.popup = null;
-                } else if (me.getElement().find('.content')[0]) {
-                    me.closeSelection();
+                if (me.popupControls) {
+                    me.popupCleanup();
                 } else {
-                    me.openSelection();
+                    const mapModule = me.getMapModule();
+                    me.popupControls = showLayerSelectionPopup(
+                        me._baseLayers,
+                        me._layers,
+                        () => me.popupCleanup(),
+                        me._showMetadata,
+                        (layer, visible, isBaseLayer) => me._setLayerVisible(layer, visible, isBaseLayer),
```

Comment:
We need to pass `this._styleSelectable` here as well to see if style selection should be shown for user.